tools/autograd/derivatives.yaml

```diff
@@ -2786,8 +2786,13 @@
   result: auto_linear
 
 - name: unbind.int(Tensor(a -> *) self, int dim=0) -> Tensor(a)[]
-  self: unbind_backward(grads, dim)
-  result: auto_linear
+  dispatch:
+    Default:
+      self: unbind_backward(grads, dim)
+      result: auto_linear
+    AutogradNestedTensor:
+      self: at::stack(grads, dim)
```

Comment:
If that worked, we would use it :p
I think that proper use of `zeros_like()` in unbind_backward should do the trick?

Reply:
@albanD I added a separate `unbind_backward_nested` to cover this since we need to store the full NT shape info to "stack" properly. My approach stores `result` (a `vector<Tensor>`) for backward, and interestingly this hit some codegen asserts assuming foreach ops when storing a vector result. If I tweak those asserts, things seem to still work. Thoughts on the right way to handle this?